app/models/episode.rb

```diff
@@ -72,6 +72,18 @@ def self.story_uri(story)
     (story.links['self'].href || '').gsub('/authorization/', '/')
   end
 
+  def self.generate_item_guid(podcast_id, episode_guid)
+    "prx_#{podcast_id}_#{episode_guid}"
+  end
+
+  def self.decode_item_guid(item_guid)
+    item_guid.sub(/^prx_[0-9]+_/, '') if item_guid&.starts_with?('prx_')
+  end
+
+  def self.find_by_item_guid(guid)
+    where(original_guid: guid).or(where(original_guid: nil, guid: decode_item_guid(guid))).first
```

Comment:
There was probably some clever SQL to generate the `prx_123_abcd` style guid in postgres?  But I opted to just decode it in Ruby.